apollo-federation/src/sources/connect/json_selection/pretty.rs

```diff
@@ -5,8 +5,11 @@
 //! pretty printing trait which is then implemented on the various sub types
```

Comment:
Do we only use this module when printing debug messages? I think we actually want to store the original instead of re-printing to look more familiar to users—so this might become test-only in the future (until/unless our dev tools add a reformat quick action 🤔).

Reply:
That's a good general question, but I don't think it's directly implicated in this PR?

Reply:
[It currently is test only](https://github.com/apollographql/router/blob/f1daee65f5aad79442e4afa526768129948a2e5b/apollo-federation/src/sources/connect/json_selection/mod.rs#L16)!

Reply:
Actually we do use it for debug response extensions, it's only the re-export that's under the `cfg` not the module itself. So the comment in your link is inaccurate (typical comment...).

But I did double-check, and it's only user-facing via the `Display` on JSONSelection types, and that's _only_ used in our debug info, so not super critical to review these bits (i.e., no performance concerns).